sisl/io/tests/test_object.py

```diff
@@ -65,22 +63,24 @@ def test_siesta_sources(self):
         pytest.importorskip("sisl.io.siesta._siesta")
 
     def test_direct_path_instantiation(self):
-        fp = Path("/tmp/shouldnotexist.1234567")
-        if fp.exists():
-            os.remove(str(fp))
-        for Sile in get_siles():
-            sile = Sile(fp, _open=False)
-            assert isinstance(sile, Sile)
-            assert not fp.exists()
+        with tempfile.TemporaryDirectory() as dir:
```

Comment:
Change to:

```python
    def test_direct_path_instantiation(self, sisl_tmp):
fp = sisl_tmp.file('sisl', 'shouldnotexist.1234657')
```
which will create a temporary file for you (and automatically delete it once the test is done.

For details, see `sisl/conftest.py`, I would suggest you to add `pathlib` to that file as well. :)

Thanks!
￼